arrow-data-source/standard/src/main/scala/com/intel/oap/spark/sql/execution/datasources/v2/arrow/ArrowScan.scala

```diff
@@ -63,4 +74,146 @@ case class ArrowScan(
   override def withFilters(partitionFilters: Seq[Expression],
                            dataFilters: Seq[Expression]): FileScan =
     this.copy(partitionFilters = partitionFilters, dataFilters = dataFilters)
+
+  // compute maxSplitBytes
+//  def maxSplitBytes(sparkSession: SparkSession,
+//                    selectedPartitions: Seq[PartitionDirectory]): Long = {
+//    // TODO: unify it with PREFERRED_PARTITION_SIZE_UPPER_BOUND.
+//    val defaultMaxSplitBytes = sparkSession.sessionState.conf.filesMaxPartitionBytes
+//    val openCostInBytes = sparkSession.sessionState.conf.filesOpenCostInBytes
+//    // val minPartitionNum = sparkSession.sessionState.conf.filesMinPartitionNum
+//    //  .getOrElse(sparkSession.leafNodeDefaultParallelism)
+//    val minPartitionNum = sparkSession.sessionState.conf.filesMinPartitionNum
+//      .getOrElse(SparkShimLoader.getSparkShims.leafNodeDefaultParallelism(sparkSession))
+//    val PREFERRED_PARTITION_SIZE_LOWER_BOUND: Long = 256 * 1024 * 1024
+//    val PREFERRED_PARTITION_SIZE_UPPER_BOUND: Long = 1024 * 1024 * 1024
+//    val totalBytes = selectedPartitions.flatMap(_.files.map(_.getLen + openCostInBytes)).sum
+//    var maxBytesPerCore = totalBytes / minPartitionNum
+//    var bytesPerCoreFinal = maxBytesPerCore
+//    var bytesPerCore = maxBytesPerCore
+//
+//    if (bytesPerCore > PREFERRED_PARTITION_SIZE_UPPER_BOUND) {
+//      // Adjust partition size.
+//      var i = 2
+//      while (bytesPerCore > PREFERRED_PARTITION_SIZE_UPPER_BOUND) {
+//        bytesPerCore = maxBytesPerCore / i
+//        if (bytesPerCore > PREFERRED_PARTITION_SIZE_LOWER_BOUND) {
+//          bytesPerCoreFinal = bytesPerCore
+//        }
+//        i = i + 1
+//      }
+//      Math.min(PREFERRED_PARTITION_SIZE_UPPER_BOUND, bytesPerCoreFinal)
+//      // Math.min(defaultMaxSplitBytes, Math.max(openCostInBytes, bytesPerCore))
+//    } else {
+//      // adjust open cost.
+//      var i = 2
+//      while (bytesPerCore < PREFERRED_PARTITION_SIZE_LOWER_BOUND) {
+//        val dynamicOpenCostInBytes = openCostInBytesFinal * i
+//        val totalBytes =
+//          selectedPartitions.flatMap(_.files.map(_.getLen + dynamicOpenCostInBytes)).sum
+//        maxBytesPerCore = totalBytes / minPartitionNum
+//        if (maxBytesPerCore < PREFERRED_PARTITION_SIZE_UPPER_BOUND) {
+//          openCostInBytesFinal = dynamicOpenCostInBytes
+//          bytesPerCoreFinal = maxBytesPerCore
+//        }
+//        i = i + 1
+//      }
+//      Math.max(PREFERRED_PARTITION_SIZE_LOWER_BOUND, bytesPerCoreFinal)
+//    }
+//  }
+
+  // This implementation is ported from spark FilePartition.scala with changes for
+  // adjusting openCost.
+  def getFilePartitions(sparkSession: SparkSession,
```

Comment:
@jackylee-ch, please put your code changes for open cost here. It should be workable.

Reply:
Okey